src/file.cr

```diff
@@ -377,6 +378,15 @@ class File < IO::FileDescriptor
     stat.size
   end
 
+  # Truncates the file to the specified length. Requires a write file descriptor
+  def truncate(length)
```

Comment:
A default makes total sense. I was thinking about another possibility, let me know what you think: the current `pos`. This way you can stick to the default for the two main (that I can think of) use cases:
* You just opened the file and want to truncate it entirely (current `pos` would be `0`)
* You just finished writing and want to truncate the rest (current `pos` would be where you finished writing)
I think this makes sense but it might have problems I'm not seeing, if you don't like it, I'll just add the `0` default, otherwise I'll implement the one I just described.

Thanks for the feedback!